app/src/main/java/com/example/youthero/MainActivity.kt

```diff
@@ -3,28 +3,44 @@ package com.example.youthero
 import android.os.Bundle
 import androidx.activity.ComponentActivity
 import androidx.activity.compose.setContent
+import androidx.navigation.NavType
+import androidx.navigation.compose.NavHost
+import androidx.navigation.compose.composable
+import androidx.navigation.compose.rememberNavController
+import androidx.navigation.navArgument
 import com.example.youthero.categories.Category
 import com.example.youthero.categories.MenuCategories
+import com.example.youthero.details.Details
 import com.example.youthero.ui.theme.YoutheroTheme
 
 class MainActivity : ComponentActivity() {
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
             val listCategories: List<Category> = listOf(
-                Category("Educatie", R.drawable.educatie),
-                Category("Cultura", R.drawable.cultura),
-                Category("Ocupatie", R.drawable.ocupatie),
-                Category("Mediu inconjurator", R.drawable.mediu_inconjurator),
-                Category("Viata sanatoasa", R.drawable.viata_sanatoasa),
-                Category("Drepturile Tinerilor", R.drawable.drepturi_tineri),
-                Category("Participare", R.drawable.participare),
-                Category("Spatii pentru tineri", R.drawable.spatii_tineri)
+                Category(getString(R.string.education), R.drawable.educatie),
+                Category(getString(R.string.culture), R.drawable.cultura),
+                Category(getString(R.string.job), R.drawable.ocupatie),
+                Category(getString(R.string.enviroment), R.drawable.mediu_inconjurator),
+                Category(getString(R.string.healthy), R.drawable.viata_sanatoasa),
+                Category(getString(R.string.rights), R.drawable.drepturi_tineri),
+                Category(getString(R.string.participation), R.drawable.participare),
+                Category(getString(R.string.spaces), R.drawable.spatii_tineri)
             )
             YoutheroTheme {
-                MenuCategories(listCategories)
-                //Ce va urma...
+                val navController = rememberNavController()
+                NavHost(navController = navController, startDestination = "categories") {
+                    composable("categories") { MenuCategories(listCategories = listCategories, navController = navController) }
```

Comment:
This is one way to do it but there a couple of potential issues that might happen when implementing it like this:

- the categories controller is very hard to test this way, as one will need to inject a navController into it.
- this also violates [the single source of truth principle](https://developer.android.com/topic/architecture#single-source-of-truth), the `navigate` method changes the internal state of the `navController`, so in this case the categories composable changes the controller internal state when it calls navigate on it. 

A better approach will be to use [state hoisting](https://developer.android.com/jetpack/compose/state#state-hoisting), read more about navigation usage [here](https://developer.android.com/jetpack/compose/navigation#nav-from-composable)

Reply:
I will try again to look intro this issue...